coral/src/app/features/topics/overview/messages/components/TopicMessageItem.tsx

```diff
@@ -0,0 +1,44 @@
+import { BorderBox, Box, Button, Typography } from "@aivenio/aquarium";
+import expandIcon from "@aivenio/aquarium/icons/chevronRight";
+import minimizeIcon from "@aivenio/aquarium/icons/chevronDown";
+import { useState, useId } from "react";
+import truncate from "lodash/truncate";
+
+type Props = {
+  offsetId: string;
+  message: string;
+};
+
+function TopicMessageItem({ offsetId, message }: Props) {
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const panelTrigger = useId();
+  const panelId = useId();
+  return (
+    <BorderBox padding={"4"} marginBottom={"4"}>
+      <Box.Flex component={"h3"}>
+        <Box marginRight="2">
+          <Button.Icon
+            id={panelTrigger}
+            icon={expanded ? minimizeIcon : expandIcon}
+            aria-label={`Expand message ${offsetId}`}
+            aria-expanded={expanded}
+            aria-controls={panelId}
+            onClick={() => setExpanded(!expanded)}
+          />
+        </Box>
+        <Box
```

Comment:
Maybe we could add a `title` prop on this `Box` so that the full message is still legible even not expanded? Maybe annoying / overkill :o

https://github.com/aiven/klaw/assets/20607294/e635a263-f8c0-4596-9f13-8bd4fbc09d19

Reply:
Hmm, let's keep this in mind. I'm thinking the same, it might be nice but might also get in the way. At least we'd have to make sure it's not displayed when the content is expanded.